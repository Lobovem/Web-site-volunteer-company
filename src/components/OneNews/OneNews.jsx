import { useLoaderData, useParams } from 'react-router-dom';
import { SliderFifth } from '../Sliders/SliderFifth/SliderFifth';
import { YoutubeVideo } from '../YoutubeVideo/YoutubeVideo';
import { oneNewsSelector } from '../../redux/slice/contentSlice';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import s from './OneNews.module.scss';
import { Loader } from '../Loader/Loader';

export const fetchOneNews = async (id) => {
  try {
    const response = await fetch(`https://base-twmn.onrender.com/news/${id}`);
    if (!response.status === 200) {
      throw new Error('Error fetching news list');
    }
    const data = await response.json();
    return data;
  } catch (error) {
    throw new Error(error.message);
  }
};

export const OneNews = () => {
  const { id } = useParams();

  // useEffect(() => {
  // dispatch(fetchOneNews(id)); // Dispatch the action to fetch news data by ID
  // }, [id]);

  //selector for simple news data
  const oneNews = useLoaderData();

  return (
    <>
      <div className={s.simpleNews}>
        <div key={oneNews.id}>
          <h1 className={s.simpleNews__title}>{oneNews.title}</h1>
          <div className={s.simpleNews__wrap}>
            <h3 className={s.simpleNews__decor}>Новини</h3>
            <div>
              <p className={s.simpleNews__text}>{oneNews.desc}</p>
            </div>
            <SliderFifth news={oneNews}></SliderFifth>
          </div>
        </div>
      </div>
      <YoutubeVideo></YoutubeVideo>
    </>
  );
};
