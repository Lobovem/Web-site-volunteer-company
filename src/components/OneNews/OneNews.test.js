import { render, screen } from '@testing-library/react';
import { OneNews } from './OneNews';
import { useDispatch, useSelector } from 'react-redux';

// jest.mock('../../api/api', () => ({
//   fetchMenu: jest.fn(),
//   fetchNews: jest.fn(),
// }));

jest.mock('react-redux', () => {
  const originalModule = jest.requireActual('react-redux');
  return {
    ...originalModule,
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

describe('OneNews', () => {
  const dispatch = jest.fn();

  test('renders OneNews component', () => {
    const oneNews = { id: 1, title: 'Новина', desc: 'News' };
    useDispatch.mockReturnValue(dispatch);
    useSelector.mockReturnValue(oneNews);

    const { container } = render(<OneNews />);
    expect(container).toMatchSnapshot();
  });

  test('search tag h3 (decor)', () => {
    render(<oneNews />);
    const h3Element = screen.getByRole('heading', { level: 1 });
    expect(h3Element).toBeInTheDocument();
  });
});
