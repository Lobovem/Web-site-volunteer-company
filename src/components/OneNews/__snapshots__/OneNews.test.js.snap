// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneNews renders OneNews component 1`] = `
<div>
  <div
    class="simpleNews"
  >
    <div>
      <h1
        class="simpleNews__title"
      >
        Новина
      </h1>
      <div
        class="simpleNews__wrap"
      >
        <h3
          class="simpleNews__decor"
        >
          Новини
        </h3>
        <div>
          <p
            class="simpleNews__text"
          >
            News
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="youtubeVideo"
  >
    <div
      class="youtubeVideo__wrapIframe"
    >
      <iframe
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
        allowfullscreen=""
        class="youtubeVideo__iframe"
        frameborder="0"
        src="https://www.youtube.com/embed/yz9zqm-TAIg"
        title="YouTube video player"
      />
    </div>
    <h2
      class="titleMain"
    >
      Допомога нашим братам та сестрам
    </h2>
    <p
      class="youtubeVideo__desc"
    >
      Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації. Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації.
    </p>
    <p
      class="youtubeVideo__desc"
    >
      За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично. Нашою першочерговою метою є вирішення нагальних проблем ЗСУ. За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки...
    </p>
  </div>
</div>
`;
